### PR TITLE
Update to bitintr 0.2

### PR DESCRIPTION
**Pull request by callym**
This just updates to bitintr 0.2, I had to recreate the trait that was used, but that's the only big change, just a lot of renaming otherwise.
Not sure if this is really needed as I'm not sure whether there are any functional differences between bitintr 0.1 and 0.2, but thought I'd send across a pull request anyway.

### CODE REVIEW

File: src/word/word.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
+//! Generic Integer traits.
+
```
Comment (gnzlbg) [inline src/word/word.rs:2]: Ah crap. Did I removed this trait from bitintr and that's why you had to recreate it?

**Comment by gnzlbg** [comment]
LGTM. What are you using bitintr for if I might ask?

File: src/word/word.rs (hunk 1 above)
Comment (callym) [reply src/word/word.rs:2]: Yeah, it was mostly adding loads of trait bounds for the stuff from bitintr that it needed.

**Comment by callym** [comment]
I'm working my way through Physically Based Rendering and they use Morton codes to create a BVH, I thought it'd probably be better to use an existing library than hoping I'd got all the bitshifting and low-level code right 😄

**Comment by gnzlbg** [comment]
I wanted to split those into their own crate, and maybe have a more user friendly "morton index" type that you can just work on (e.g. do "pointwise arithmetic", query the "level" within the tree, etc.), with the aim of building a quad-tree library on top of it (I have never used morton codes with BVH, does the BVH creation algorithm use an octree ?)

**Comment by callym** [comment]
They show how quite a few BVH construction algorithms work, the one that uses Morton indices is a HLBVH that builds the tree in parallel and uses the Morton codes to do quick radix sorting on the shapes, not sure if that's similar to an octree?
The code is [here](https://github.com/callym/pbrst/blob/45c291ff9e833c0731d37e218bf30761791d18b9/pbrt/src/aggregate/bvh.rs#L321) if you're interested in it - can't guarantee it's the easiest or best code to read through as I'm still new to a lot of these concepts.

**Comment by gnzlbg** [comment]
Thanks, I always wanted to go through pbrt. I guess it is time :D

**Pull request merged**